src/ttsim/plot_dag.py

```diff
@@ -123,16 +191,93 @@ def plot_full_interface_dag(output_path: Path) -> None:
             "The full interface DAG should include all root nodes but requires inputs:"
             f"\n\n{format_list_linewise(args.keys())}"
         )
-    fig = _plot_dag(dag=dag, title="Full Interface DAG")
-    if output_path.suffix == ".html":
-        fig.write_html(output_path)
+    return _plot_dag(dag=dag, title="Full Interface DAG")
+
+
+def all_targets_from_namespace(
+    inputs_for_main: dict[str, Any],
+) -> list[str]:
+    """Get all targets from the original policy objects / params functions."""
+    return main(
+        inputs=inputs_for_main,
+        targets=["targets__qname"],
+    )["targets__qname"]
+
+
+def specialized_environment_for_targets(
```

Comment:
Let's see whether we can recycle more code here. The stuff in `specialized_environment` is the most complex part of the code lots of internal (and-probably-still-changing...) logic. Let's chat about that!